firmware/Core/Src/telecommands/telecommand_args_helpers.c

```diff
@@ -83,6 +83,86 @@ uint8_t TCMD_extract_uint64_arg(const char *str, uint32_t str_len, uint8_t arg_i
     return 0;
 }
 
+/// @brief Extracts the longest substring of integer characters, starting from the beginning of the
+///     string, to a maximum length or until the first non-integer character is found.
+/// @param str Input string, starting with an integer
+/// @param str_len Max length of the input string
+/// @param result Pointer to the result
+/// @return 0 if successful, 1 if the string is empty, 2 if the string does not entirely encapsulate an integer
+uint8_t TCMD_ascii_to_int64(const char *str, uint32_t str_len, int64_t *result) {
+    if (str_len == 0) {
+        return 1;
+    }
+    uint64_t temp_result = 0;
```

Comment:
Temp result is unsigned? You multiply by negative 1 later.

Reply:
Thanks!